### ISSUE TITLE
[Feature Request] Replace the behavior of "Download PDF" with "View PDF" in a modal window

### ISSUE THREAD

**Issue by Verony-makesIT** [InvoicePlane/InvoicePlane#925]
In several places (in Quotes and Invoices, among others) you can find the "Options" dropdown menu. 
Here you can select the menu item "Download PDF." The application then jumps to a new tab window and the corresponding Quote or Invoice pdf is displayed. 

## Expected Behavior
Choose the menu item "View PDF" which opens a modal window (with the PDF) on top of the current page

## Current Behavior
Menu item "Download PDF" opens a new tab window, jump to it and show the PDF there

## Screenshots
![image](https://user-images.githubusercontent.com/30659226/218269055-868b4420-ecbc-4f31-b332-c78b21d846ce.png)

## Possible Solution
Replace the behavior of "Download PDF" with "View PDF" and show PDFs anywhere in the application in a modal window

## Context (Environment)
Each time you click on this menu item, a new tab window will open where the pdf is displayed. 
This is very confusing and you may have multiple tab windows open with different pdf versions of the same Quote or Invoice as a result.

**Comment by gaelicwinter** [comment]
This would cause a problem when I'm working on a quote or invoice and I
want to see what it now looks like.  Currently I can let the quote open in
the new tab, return to the first tab and edit the quote, then refresh the
second tab and see the changes.  With this proposed change, I'd have to
constantly be closing and re-opening the modal which I think would be
troublesome.
=============================================

**Comment by clockwiseq** [comment]
I have to agree with @gaelicwinter.  I believe the current functionality suffices and maybe we need to add a second button for "preview" and that would perform the action @Verony-makesIT has described above.

**Comment by Verony-makesIT** [comment]
If I follow @gaelicwinter procedure he performs following steps:
1. choose menu item "Download PDF"
2. view pdf in open tab
3. switch back to IP quote/invoice
4. make adjustments
5. switch back to the tab
6. refresh tab and view adjustment
7. close tab if everything is ok
8. switch back to the IP tab

My suggestion:
1. choose menu item "View PDF"
2. view pdf in open modal window
3. click to close this modal window
4. make adjustments
5. select menu item and view modification
6. close modal window if everything is ok

Conclusion: 
To do the same thing, in my proposal there are actually **2 fewer steps to perform** and you keep working on the same web page. For many people, this will feel more natural and be more pleasant to work with. 
I created this because it is precisely difficult for me to do this the "IP way".

Can't we leave the choice to the user which option they want to use by adding an additional setting [**Download PDF in tab**] or [**View PDF in modal**]?

**Comment by nielsdrost7** [comment]
My 2 cents:
Open "Download PDF", it will open in new tab.
Switch back to old tab
Move new tab to a "new window"
Place the windows side-by-side
Adjust on 1 side an view on the other side.
No need to change anything ... yet.

Don't delete options, but add new ones: View PDF, or like clockwiseq said: `Preview PDF`, that one can open in a modal.
Getting those PDFS to view correctly in some browsers is a terrible task, stay away from it, is what I would say.

To test the behaviour, don't rely on your own PC, there's an application called 'Browserstack' which will prepare your windows types, Mac types and several exotic browsers

**Comment by Verony-makesIT** [comment]
I think there is a misunderstanding somewhere.
My intention is not to remove or replace an option but to give the user a choice between the visibility of the 2 options via a setting in IP. 
Example: 
- By default the setting is off and only the menu choice "Download PDF" is visible. 
- If the setting is on then the menu choice "Preview PDF" is (additionally?) visible in the 'Options menus'. 

Regarding your comment about viewing pdfs in the browser, I just want to point out that I do not want to introduce new techniques in IP but only use the same modal functionality that is present in IP since years. E.g. modal create invoice, add_invoice_tax, etc.. So nothing new, just instead of some text and form fields I show the pdfs in an iframe in the body of the modal. The pdfs are rendered with their own internal pdf browser engine. 
I don't think that should be a problem for any browser.
